massa-bootstrap/src/client.rs

```diff
@@ -518,7 +497,52 @@ pub async fn get_state(
             };
 
             info!("Bootstrap from server {} failed. Your node will try to bootstrap from another server in {}.", addr, format_duration(bootstrap_config.retry_delay.to_duration()).to_string());
-            sleep(bootstrap_config.retry_delay.into()).await;
+            std::thread::sleep(bootstrap_config.retry_delay.into());
         }
     }
 }
+
+fn get_genesis_state(
+    final_state: Arc<RwLock<FinalState>>,
+) -> Result<GlobalBootstrapState, BootstrapError> {
+    massa_trace!("bootstrap.lib.get_state.init_from_scratch", {});
+    // init final state
+    {
+        let mut final_state_guard = final_state.write();
+        // load ledger from initial ledger file
+        final_state_guard
+            .ledger
+            .load_initial_ledger()
+            .map_err(|err| {
+                BootstrapError::GeneralError(format!("could not load initial ledger: {}", err))
+            })?;
+        // create the initial cycle of PoS cycle_history
+        final_state_guard.pos_state.create_initial_cycle();
+    }
+    Ok(GlobalBootstrapState::new(final_state))
+}
```

Comment:
@Leo-Besancon  This is a pure refactor to move code into a helper function. Can you please confirm that it shouldn't interfere with your restart contributions?

Reply:
Yes it's not a problem at all, let's see who merges first to see who will handle the small conflict 😁

Reply:
I've merged in a recent testnet 21. could you please double check that I haven't clobbered any of your [client.rs](https://github.com/massalabs/massa/pull/3745/files#diff-20eadb4ee40e0071fec6af0f8bcf8e8ce658f07335590ce21173b8099ad259c1) work?

Reply:
...also other parts of the bootstrap module as well, come to think of it.

Reply:
...latest version of PR reverts this change